### PR TITLE
Adding support for GlobalForwardingRule from CAI

### PR DESCRIPTION
**Pull request by red2k18**

### CODE REVIEW

File: tests/services/inventory/test_data/additional_cai_resources.dump [hunk 1]
```diff
@@ -12,6 +12,8 @@
...
+# Compute GlobalForwardingRule
+{"asset_type":"compute.googleapis.com/GlobalForwardingRule","name":"//compute.googleapis.com/projects/project1/regions/us-central1/global/forwardingRules/lb-1","resource":{"version": "v1","discovery_document_uri":"https://www.googleapis.com/discovery/v1/apis/compute/v1/rest","discovery_name": "GlobalForwardingRule","parent":"//cloudresourcemanager.googleapis.com/projects/94226340476","data":{"IPAddress":"172.16.1.2","IPProtocol":"TCP","creationTimestamp":"2017-05-05T12:00:01.000-07:00","description":"","id":"1111","kind":"compute#forwardingRule","loadBalancingScheme":"EXTERNAL","name":"lb-1","portRange":"80-80","region":"https://www.googleapis.com/compute/v1/projects/project1/regions/us-central1","selfLink":"https://www.googleapis.com/compute/v1/projects/project1/regions/us-central1/global/forwardingRules/lb-1",      "networkTier": "PREMIUM","target":"https://www.googleapis.com/compute/v1/projects/project1/regions/us-central1/targetHttpProxies/lb-1-target-proxy"},"parent":"//cloudresourcemanager.googleapis.com/projects/1041"}}
```
Comment (ahoying) [inline tests/services/inventory/test_data/additional_cai_resources.dump]: Please change project ID of parent from 94226340476 to 1042 or the crawler will not discover this resource. Remove the second parent entry.

Set the id to 9161 and note in the comment that the ID reserved space is (id 916x). All IDs across all resources under test need to be unique, so we used reserved space for each resource to ensure that is the case.

Comment (ahoying) [inline tests/services/inventory/test_data/additional_cai_resources.dump]: Remove blank line.

Comment (ahoying) [reply tests/services/inventory/test_data/additional_cai_resources.dump]: Once this resource is discovered correctly, the crawler test will fail until the resource count is increased for the forwarding rules resource.

File: tests/services/inventory/update_cai_dumps.py [hunk 2]
```diff
@@ -256,6 +256,11 @@ def forwardingrule(item):
     return _create_compute_asset(item, 'compute.googleapis.com/ForwardingRule')
 
 
+def globalforwardingrule(item):
```
Comment (ahoying) [inline tests/services/inventory/update_cai_dumps.py]: This file shouldn't need to be changed, this is needed when going from the mock API to the CAI dump, not the other way around.

File: configs/server/forseti_conf_server.yaml.in [hunk 3]
```diff
@@ -143,6 +143,7 @@ inventory:
         #    - compute.googleapis.com/Disk
         #    - compute.googleapis.com/Firewall
         #    - compute.googleapis.com/ForwardingRule
+        #    - compute.googleapis.com/GlobalForwardingRule
```
Comment (ahoying) [inline configs/server/forseti_conf_server.yaml.in]: Also update the sample config file.

Comment (red2k18) [reply configs/server/forseti_conf_server.yaml.in]: Done.

Comment (red2k18) [reply configs/server/forseti_conf_server.yaml.in]: Done

File: tests/services/inventory/update_cai_dumps.py (hunk 2 above)
Comment (red2k18) [reply tests/services/inventory/update_cai_dumps.py]: Gotcha. Reverted changes.

File: tests/services/inventory/test_data/additional_cai_resources.dump (hunk 1 above)
Comment (red2k18) [reply tests/services/inventory/test_data/additional_cai_resources.dump]: There's no blank line.

Comment (red2k18) [reply tests/services/inventory/test_data/additional_cai_resources.dump]: Thanks so much for the explanation. Updated project ID and id and incremented expected count for forwarding rules resource. But the test fails as the resource is not discovered correctly. Am I missing something?

File: tests/services/inventory/crawling_test.py [hunk 4]
```diff
@@ -290,7 +290,7 @@ def test_crawling_from_composite_root(self):
             'disk': {'resource': 3},
             'firewall': {'resource': 3},
             'folder': {'iam_policy': 2, 'resource': 2},
-            'forwardingrule': {'resource': 1},
```
Comment (ahoying) [inline tests/services/inventory/crawling_test.py]: This one won't change as the resource is not included in the project getting tested with the composite root.

File: tests/services/inventory/crawling_test.py [hunk 5]
```diff
@@ -490,7 +490,6 @@ def test_crawl_cai_api_polling_disabled(self):
             'compute_router': {'resource': 1},
             'compute_sslcertificate': {'resource': 1},
             'compute_targethttpproxy': {'resource': 1},
-            'compute_targethttpsproxy': {'resource': 1},
```
Comment (ahoying) [inline tests/services/inventory/crawling_test.py]: This should not be removed

The forwardingrule entry below should be increased to 2, that's where the new global forwarding rule shows up.

You also need to add forwardingrule with a resource count of 2 to the test_cai_crawl_to_memory test.

File: tests/services/inventory/crawling_test.py (hunk 4 above)
Comment (red2k18) [reply tests/services/inventory/crawling_test.py]: Done

File: tests/services/inventory/crawling_test.py (hunk 5 above)
Comment (red2k18) [reply tests/services/inventory/crawling_test.py]: Done

**Comment by ahoying** [comment]
Approved pending resolving remaining travis issues. Thanks!

**Pull request merged**